src/D2L.CodeStyle.Analyzers/Language/OnlyVisibleToAnalyzer.Model.cs

```diff
@@ -126,6 +133,21 @@ INamedTypeSymbol onlyVisibleToTypeAttribute
 
 				return null;
 			}
+
+			private static INamedTypeSymbol GetTypeOrUnboundGenericType( INamedTypeSymbol type ) {
+
+				if( !type.IsGenericType ) {
+					return type;
+				}
+
+				// typeof maps to an unbound generic type
+				if( type.IsUnboundGenericType ) {
+					return type;
+				}
+
+				// GetTypeByMetadataName maps to a bound generic type of T
+				return type.ConstructUnboundGenericType();

```

Comment:
I expected `type.OriginalDefinition` to be enough, but I suppose this is recursive on the containing type where the first is not?

Reply:
I tried doing `type.ConstructUnboundGenericType().ConstructUnboundGenericType().ConstructUnboundGenericType()` but it never got to the unbound.

Reply:
Tried the other direction and that works:

> visibleToType
> {TestCases.AllowedGenericCaller<>}
>     NullableAnnotation: None

> visibleToType.OriginalDefinition
> {TestCases.AllowedGenericCaller<T>}
>     NullableAnnotation: None